unit_test/internal/worker/test_params_dict.py

```diff
@@ -1,13 +1,158 @@
 
 
 class TestModule(unittest.TestCase):
+    def test__convert_params_dict_to_struct__list_bool_float_int_datetime_timedelta(self) -> None:
+        # Arrange
+        params_dict: ParamsDict = {
+            "list": [1, 2, 3],
+            "bool": True,
+            "str": "some-str",
+            "float": 2.0,
+            "int": 3,
+            "datetime": datetime.fromisoformat("2019-01-01T01:00:00"),
+            "timedelta": timedelta(hours=1, seconds=15),
+        }
+
+        # Act
+        converted = convert_params_dict_to_struct(params_dict)
+
+        # Assert
+        expected_converted = {
+            "list": [1.0, 2.0, 3.0],
+            "bool": True,
+            "str": "some-str",
+            "float": 2.0,
+            "int": 3.0,
+            "datetime": datetime.fromisoformat("2019-01-01T01:00:00").timestamp(),
+            "timedelta": 3615.0,
+        }
+
+        self.assertEqual(json_format.MessageToDict(converted), expected_converted)
+
+    def test__parse_workflow_config_parameter__list(self) -> None:
+        # Arrange
+        workflow_config: ParamsDict = {"some-key": [1.0, 2.0, 3.0]}
+        field_key: str = "some-key"
+        expected_type: Type[list] = list
+        default_value: list[float] = []
+
+        # Act
+        param = parse_workflow_config_parameter(
+            workflow_config, field_key, expected_type, default_value
+        )
+
+        # Assert
+        expected_param = [1.0, 2.0, 3.0]
+        self.assertEqual(param, expected_param)
+
+    def test__parse_workflow_config_parameter__bool(self) -> None:
+        # Arrange
+        workflow_config = {"some-key": True}
+        field_key = "some-key"
+        expected_type = bool
+        default_value = False
+
+        # Act
+        param = parse_workflow_config_parameter(
+            workflow_config, field_key, expected_type, default_value
+        )
+
+        # Assert
+        self.assertTrue(param)
+
+    def test__parse_workflow_config_parameter__str(self) -> None:
+        # Arrange
+        workflow_config = {"some-key": "some-str"}
+        field_key = "some-key"
+        expected_type = str
+        default_value = "default"
+
+        # Act
+        param = parse_workflow_config_parameter(
+            workflow_config, field_key, expected_type, default_value
+        )
+
+        # Assert
+        expected_param = "some-str"
+        self.assertEqual(param, expected_param)
+
+    def test__parse_workflow_config_parameter__float(self) -> None:
+        # Arrange
+        workflow_config = {"some-key": 2.0}
+        field_key = "some-key"
+        expected_type = float
+        default_value = 1
+
+        # Act
+        param = parse_workflow_config_parameter(
+            workflow_config, field_key, expected_type, default_value
+        )
+
+        # Assert
+        expected_param = 2.0
+        self.assertEqual(param, expected_param)
+
+    def test__parse_workflow_config_parameter__int(self) -> None:
```

Comment:
This one is also similar to `test__parse_workflow_config_parameter__key_available_expect_int_but_get_float`

Reply:
This one is different. In `test__parse_workflow_config_parameter__key_available_expect_int_but_get_float` we receive a float but expect an int. In `test__parse_workflow_config_parameter__int` we receive an int (a float with `.0` as protobuf doesn't support integers) and expect an int. 

We use `test__parse_workflow_config_parameter__key_available_expect_int_but_get_float` to test the warning/rounding if it works correctly as the input float is not a rounded `int`. It ends with `.4` and not `.0`.

Reply:
Yes, that's right.